<?xml version="1.0"?>

<launch>

  <!-- Start Mocap system transmission -->
  <include file="$(find arm_hand_capture)/launch/load_vrpn.launch">
    <arg name="server" value="192.168.66.1" />
  </include>

  <include file="$(find rosserial_server)/launch/socket.launch" />


  <!-- Synchronize the mocap data and glove data(and transform to z-up frames) -->
  <node name="sync_message_topic" pkg="arm_hand_capture" type="sync_message_topic" output="screen" />


  <!-- Unpack the synced data and broadcast the tf, for Visualization -->
  <node name="unpack_synced_message" pkg="arm_hand_capture" type="unpack_synced_message" /> <!-- output="screen" /-->

  
  <!-- Transform the synced data to match the configuration of UR5 -->
  <!--node name="synced_message_to_ur5" pkg="arm_hand_capture" type="synced_message_to_ur5" /-->
  <node name="synced_message_to_yumi" pkg="arm_hand_capture" type="synced_message_to_yumi" />


  <!-- Start Rviz for visualization -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find arm_hand_capture)/rviz/mocap_tf.rviz" />



</launch>
